presto-hive/src/main/java/io/prestosql/plugin/hive/HiveType.java

```diff
@@ -254,17 +256,15 @@ private static TypeSignature getTypeSignature(TypeInfo typeInfo)
                 if (fieldTypes.size() != fieldNames.size()) {
                     throw new PrestoException(HiveErrorCode.HIVE_INVALID_METADATA, format("Invalid Hive struct type: %s", typeInfo));
                 }
-                ImmutableList.Builder<TypeSignatureParameter> typeSignatureBuilder = ImmutableList.builder();
-                for (int i = 0; i < fieldTypes.size(); i++) {
-                    typeSignatureBuilder.add(namedField(
-                            // Lower case the struct field names.
-                            // Otherwise, Presto will refuse to write to columns whose struct type has field names containing upper case characters.
-                            // Users can't work around this by casting in their queries because Presto parser always lower case types.
-                            // TODO: This is a hack. Presto engine should be able to handle identifiers in a case insensitive way where necessary.
-                            fieldNames.get(i).toLowerCase(Locale.US),
-                            getTypeSignature(fieldTypes.get(i))));
-                }
-                return rowType(typeSignatureBuilder.build());
+                return rowType(Streams.zip(
```

Comment:
Previous comments, from [here](https://github.com/prestosql/presto/pull/5695#discussion_r531097603):

> > `Streams.zip` does not necessarily improve readability, because i need to think what's the behavior when zipped streams have different lengths. I'd suggest dropping this change.
>
> In this case, the lines above guarantee that the streams are the same size (because the two lists represents pairs of values). (This might be clearer with the comments above moved to the next line.)
>
> (I actually find the indexed for-loop slightly less readable, because there's a moment of confusion before I parse why it's indexed instead of a for-each loop.)